hibernate-core/src/main/java/org/hibernate/boot/models/xml/internal/XmlAnnotationHelper.java

```diff
@@ -479,6 +482,27 @@ public static void applyNationalized(
 		memberDetails.applyAnnotationUsage( HibernateAnnotations.NATIONALIZED, xmlDocumentContext.getModelBuildingContext() );
 	}
 
+	public static void applyGenerated(
+			GenerationTiming timing,
+			MutableMemberDetails memberDetails,
+			XmlDocumentContext xmlDocumentContext) {
+		if ( timing == null ) {
+			return;
+		}
+
+		EnumSet<EventType> eventTypes = timing.getEventTypes();
+		if ( eventTypes == null ) {
+			return;
+		}
+
+		final GeneratedAnnotation generatedAnn = (GeneratedAnnotation) memberDetails.applyAnnotationUsage(
+				HibernateAnnotations.GENERATED,
+				xmlDocumentContext.getModelBuildingContext()
+		);
+
+		generatedAnn.event( eventTypes.toArray( new EventType[0] ) );
```

Comment:
Why not allocate the right size straight away?

Reply:
As you can read in the articles, it seems that the zeroing of the memory can't be avoided when pre-allocating an array, whereas the intrinsic array copyOf seems to avoid that.